data-publisher/src/main/kotlin/no/nav/amt/tiltak/data_publisher/DataPublisherService.kt

```diff
@@ -0,0 +1,151 @@
+			publishRepository.set(id, DataPublishType.DELTAKERLISTE, currentData.digest())
+		}
+	}
+
+	private fun publishDeltaker(id: UUID, forcePublish: Boolean = false) {
+		val currentData = DeltakerPublishQuery(template).execute(id)
+
+		if (currentData == null) {
+			val key = id.toString().toByteArray()
+			val value = "".toByteArray()
+			val record = ProducerRecord(kafkaTopicProperties.amtDeltakerTopic, key, value)
+			logger.info("Legger inn Tombstone på DELTAKER med id $id")
+			kafkaProducerClient.sendSync(record)
+		}
+
+		else if (forcePublish || !publishRepository.hasHash(id, DataPublishType.DELTAKER, currentData.digest())) {
+			val key = id.toString().toByteArray()
+			val value = JsonUtils.toJsonString(currentData).toByteArray()
+			val record = ProducerRecord(kafkaTopicProperties.amtDeltakerTopic, key, value)
+			logger.info("Republiserer DELTAKER med id $id")
+			kafkaProducerClient.sendSync(record)
+			publishRepository.set(id, DataPublishType.DELTAKER, currentData.digest())
+		}
+
+	}
+
+	private fun publishArrangor(id: UUID, forcePublish: Boolean = false) {
+		val currentData = ArrangorPublishQuery(template).execute(id)
+
+		if (forcePublish || !publishRepository.hasHash(id, DataPublishType.ARRANGOR, currentData.digest())) {
+			val key = id.toString().toByteArray()
+			val value = JsonUtils.toJsonString(currentData).toByteArray()
+			val record = ProducerRecord(kafkaTopicProperties.amtArrangorTopic, key, value)
+			logger.info("Republiserer ARRANGOR med id $id")
+			kafkaProducerClient.sendSync(record)
+			publishRepository.set(id, DataPublishType.ARRANGOR, currentData.digest())
+		}
+	}
+
+	private fun publishArrangorAnsatt(id: UUID, forcePublish: Boolean = false) {
+		val currentData = ArrangorAnsattPublishQuery(template).execute(id)
+
+		if (forcePublish || !publishRepository.hasHash(id, DataPublishType.ARRANGOR_ANSATT, currentData.digest())) {
+			val key = id.toString().toByteArray()
+			val value = JsonUtils.toJsonString(currentData).toByteArray()
+			val record = ProducerRecord(kafkaTopicProperties.amtArrangorAnsattTopic, key, value)
+			logger.info("Republiserer ARRANGOR_ANSATT med id $id")
+			kafkaProducerClient.sendSync(record)
+			publishRepository.set(id, DataPublishType.ARRANGOR_ANSATT, currentData.digest())
+		}
+	}
+
+	private fun publishEndringsmelding(id: UUID, forcePublish: Boolean = false) {
+		val currentData = EndringsmeldingPublishQuery(template).execute(id)
+
+		if (forcePublish || !publishRepository.hasHash(id, DataPublishType.ENDRINGSMELDING, currentData.digest())) {
+			val key = id.toString().toByteArray()
+			val value = JsonUtils.toJsonString(currentData).toByteArray()
+			val record = ProducerRecord(kafkaTopicProperties.amtArrangorAnsattTopic, key, value)
+			logger.info("Republiserer ENDRINGSMELDING med id $id")
+			kafkaProducerClient.sendSync(record)
+			publishRepository.set(id, DataPublishType.ENDRINGSMELDING, currentData.digest())
+		}
+	}
+
+	private fun publishBatch(
```

Comment:
Denne metoden synes jeg er vanskelig å lese, og vanskelig å skjønne hva gjør (navnet på selve metoden er ok og forklarende, men resten av metoden blir for generisk til at jeg forstår hva den gjør). Går det an å gjøre den litt enklere? Og kanskje bruke mer beskrivende navn? For eksempel forstår ikke jeg hva en idProvider er/gjør 😅